source/command-logging-and-monitoring/tests/logging/redacted-commands.yml

```diff
@@ -0,0 +1,460 @@
+description: "redacted-commands"
+
+schemaVersion: "1.11"
+
+runOnRequirements:
+  - minServerVersion: "5.0"
+    auth: false
+
+createEntities:
+  - client:
+      id: &client client
+      observeLogMessages:
+        command: debug
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName logging-redaction-tests
+
+tests:
+  - description: "authenticate"
+    operations:
+      - name: runCommand
+        object: *database
+        arguments:
+          commandName: authenticate
+          command:
+            authenticate: 1
+            mechanism: "MONGODB-X509"
+            user: "CN=myName,OU=myOrgUnit,O=myOrg,L=myLocality,ST=myState,C=myCountry"
+            db: "$external"
+        # An authentication error is expected, but we want to check that the
+        # CommandStartedEvent is redacted
+        expectError:
+          isError: true
+    expectLogMessages:
+      - client: *client
+        messages:
+          - level: debug
+            component: command
+            data:
+              message: "Command started"
+              databaseName: *databaseName
+              commandName: authenticate
+              command:
+                $$matchAsDocument: {}
+          - level: debug
+            component: command
+            hasFailure: false
```

Comment:
Should all of the `hasFailure: false` in this file be `hasFailure: true`?

Reply:
hasFailure is meant to test that driver redacts the error for sensitive commands:

> ``hasFailure``: Optional boolean. When this field is set to ``true``, 
  the test runner MUST assert that the log message has a ``failure`` value
  in its attached data, and that the failure has not been redacted in any
  way. When this field is set to ``false``, the test runner MUST assert
  that the log message either has no ``failure`` value at all or that it
  has a redacted failure, e.g. an empty string or object. 
  The exact form of those assertions will vary based on the driver's chosen
  representations for failures and redacted failures in log messages.
  Currently, this assertion is only relevant for "command failed" log messages.

 some context on the redaction in this [comment](https://github.com/mongodb/specifications/pull/1303#discussion_r967396950).

since we want to redact the failure for all of these commands,  I think `false` is the right value.

Reply:
ok, I'm going to leave a comment on the `hasFailure` definition, which I now have some concerns about, and resolve this one.